### PR TITLE
Add configurable shortened event time labels (`shorten_event_times`)

### PR DESCRIPTION
**Pull request by superdingo101**
### Motivation
- Provide an option to display more compact event time labels in various calendar views to improve readability and save space.
- Allow users to opt into shortened formatting for both 12-hour and 24-hour locales while preserving existing behavior by default.

### Description
- Introduces a new configuration option `shorten_event_times` (default `false`) and exposes it in the editor UI and `getStubConfig` via `shorten_event_times` and a checkbox in the editor.
- Adds helper formatting functions: `uses24HourEventTime`, `isWholeHour`, `formatLocalizedHour`, `formatShort12HourEventTime`, `formatShort24HourEventTime`, `formatEventTime`, and `formatEventTimeRange` to implement compact rules for whole-hour and mixed-minute ranges.
- Replaces direct calls to `formatTime` for event time labels across agenda, schedule, week-compact, month, modal, and other render paths with `formatEventTime` or `formatEventTimeRange` to apply the new compacting logic when enabled.
- Updates unit tests to cover default behavior and the new compact formatting in 12-hour and 24-hour modes, range edge cases, and all-day event handling.

### Testing
- Ran the unit test suite (`skylight-calendar-card.test.js`) including new tests for `shorten_event_times`; all tests passed.
- Added tests: `shorten_event_times defaults to unchanged event time formatting`, `shorten_event_times removes minutes from whole-hour 12-hour event times`, `shorten_event_times compacts whole-hour 24-hour event times`, `shorten_event_times compacts whole-hour 24-hour ranges`, `shorten_event_times preserves needed minutes in mixed 24-hour ranges`, and `shorten_event_times leaves all-day event labels unaffected`, and they all succeeded.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a1af199efe48331b479f20ef89624ee)

### CODE REVIEW

File: skylight-calendar-card.js [hunk 1]
```diff
@@ -7480,6 +7481,72 @@ class SkylightCalendarCard extends HTMLElement {
...
+    const formatter = new Intl.DateTimeFormat(this.getLocale(), this.getTimeFormatOptions());
+    return formatter.resolvedOptions().hour12 === false;
+  }
+
+  isWholeHour(date) {
+    return date.getMinutes() === 0 && date.getSeconds() === 0 && date.getMilliseconds() === 0;
+  }
+
+  formatLocalizedHour(date) {
+    return new Intl.NumberFormat(this.getLocale(), { useGrouping: false }).format(date.getHours());
```
[the remark this answers is not in the document]
Comment (superdingo101) [reply skylight-calendar-card.js:7500]: @codex address that feedback
